src/Titon/Context/Definition/Definition.hh

```diff
@@ -0,0 +1,145 @@
+<?hh // strict
+/**
+ * @copyright   2010-2015, The Titon Project
+ * @license     http://opensource.org/licenses/bsd-license.php
+ * @link        http://titon.io
+ */
+
+namespace Titon\Context\Definition;
+
+use Closure;
+use Titon\Context\Depository;
+
+/**
+ * A definition is an object that defines how an object or callable should be
+ * created when being called or instantiated from the depository object.
+ *
+ * @package Titon\Context\Definition
+ */
+abstract class Definition
```

Comment:
Should follow the namespace organization: https://github.com/titon/framework/blob/master/docs/en/support/conventions.md#namespace-organization